addons/sale/models/product_template.py

```diff
@@ -93,3 +145,325 @@ def get_import_templates(self):
+
+        :param pricelist: `product.pricelist` the pricelist to use
+            (can be none, eg. from SO if no partner and no pricelist selected)
+
+        :param reference_product: `product.product` if no combination and
+            not product_id are given, it will try to find the first possible
+            combination, taking into account reference_product for the exclusion
+            rules (if specified).
+
+        :return: dict with product/combination info:
+
+            - product_id: the variant id matching the combination (if it exists)
+
+            - product_template_id: the current template id
+
+            - display_name: the name of the combination
+
+            - price: the computed price of the combination, take the catalog
+                price if no pricelist is given
+
+            - list_price: the catalog price of the combination, but this is
+                not the "real" list_price, it has price_extra included (so
+                it's actually more closely related to `lst_price`), and it
+                is converted to the pricelist currency (if given)
+
+            - has_discounted_price: True if the pricelist discount policy says
+                the price does not include the discount and there is actually a
+                discount applied (price < list_price), else False
+        """
+        self.ensure_one()
+
+        quantity = self._context.get('quantity', add_qty)
+        context = dict(self._context, quantity=quantity, pricelist=pricelist.id if pricelist else False)
+        product_template = self.with_context(context)
+
+        combination = combination or product_template.env['product.template.attribute.value']
+
+        if not product_id and not combination:
+            combination = product_template._get_first_possible_combination(reference_product)
+
+        if product_id and not combination:
+            product = product_template.env['product.product'].browse(product_id)
+        else:
+            # no_variant attributes can still be on a variant if their type has
+            # been changed after the creation of the variant
+            product = product_template._get_variants_for_combination(combination, exclude_no_variant=False)[:1]
+            if not product:
+                product = product_template._get_variants_for_combination(combination)[:1]
+
+        if product:
+            # We need to add the price_extra for the attributes that are not
+            # in the variant, typically those of type no_variant, but it is
+            # possible that a no_variant attribute is still in a variant if
+            # the type of the attribute has been changed after creation.
+            product = product.with_context(
+                no_variant_attributes_price_extra=[ptav.price_extra or 0.0
+                    for ptav in combination.filtered(
+                        lambda ptav: ptav not in product.product_template_attribute_value_ids
```

Comment:
@seb-odoo Maybe change the filter to "ptav.price_extra and ptav not in ..." to avoid (useless) 0.0 prices in the result ?

Reply:
ok

Reply:
In master we could also directly pass the sum instead of the array.

Reply:
Change pushed. I also changed the no_variant_attributes_price_extra on sale order because it didn't have the same filter.